slapo/model_dialect/deepspeed/engine.py

```diff
@@ -0,0 +1,25 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+from __future__ import annotations
+
+from ..registry import register_model_dialect
+from ...logger import get_logger, INFO
+
+logger = get_logger("DS-Engine", INFO)
+
+
+@register_model_dialect("deepspeed", "runtime_engine")
+def init_ds_engine(model, **kwargs):
+    """Initialize the DeepSpeed engine."""
+    import deepspeed
+
+    if "config" not in kwargs:
+        raise ValueError("DeepSpeed config not provided.")
+
+    # pylint: disable=unbalanced-tuple-unpacking
+    model, optimizer, _, _ = deepspeed.initialize(
+        model=model,
+        config=kwargs["config"],
+        model_parameters=[p for p in model.parameters() if p.requires_grad],
```

Comment:
The original implementation also doesn't expose this API. It's in the `.build` function, right?

Reply:
yes. but we would need to expose this for the internal testing. we can use `[p for p in model.parameters() if p.requires_grad]` as the default one so that we don't need modify all the scripts but we need to add model_parameters to the api so users can modify it.

Reply:
Per offline discussion, we will expose this in future PRs, because we will change the model during `build`, so user-specified parameter list may be out of date.